kernels/type_selector.h

```diff
@@ -83,6 +83,22 @@ template <> struct TypeSelector<ValueType::int1, ValueType::int32> {
   static constexpr bool IS_DOWNCAST_OP = false;
 };
 
+template <> struct TypeSelector<ValueType::float8e4m3, ValueType::float32> {
+#if (!defined(__GFX9__) && !defined(__GFX11__) && __CUDA_ARCH__ < 890)
```

Comment:
```suggestion
#if (!defined(__gfx940__) && !defined(__gfx941__) && !defined(__gfx942__) && !defined(__GFX12__) && __CUDA_ARCH__ < 890)
```

GFX9 is too broad, that includes CDNA1/2 as well. GFX11 is RDNA3, should be GFX12 for RDNA4. 
I'm not sure whether the && part works for CUDA vs HIP, is e.g. <undefined macro> < 890 true or false? Perhaps it should be split into two separate checks depending on whether it's compiling for CUDA or HIP?